### ISSUE TITLE
Allow port 0

### ISSUE THREAD

**Issue by czaplicki** [junegunn/fzf#3200]
- [X] I have read through the manual page (`man fzf`)
- [X] I have the latest version of fzf
- [X] I have searched through the existing issues

## Info

- OS
    - [X] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Etc.
- Shell
    - [ ] bash
    - [X] zsh
    - [ ] fish

## Problem / Steps to reproduce

```sh
fzf --listen 0
```

On Linux one can usually specify port 0 to ask the kernel for an available port.
I tried it but it didn't work (but no error message), so I looked in the
source code. There I saw that if port was set to 0 abort without error, so I
looked and it turned out that port's default value was zero (makes sense why it
silently aborted). But then I looked at the std net pkg, and saw that port 0 is
actually supported and not only on Linux but seems to be on any platform
(as there is no mention of it being platform dependent).

### tldr
Would be great to have access to port 0, as it would make it way easier to
write scripts that could be run simultaneously without having the ports clashing.

### Side note:
Saw that port was of type int, so a fast and easy solution to support it, would
be to just change the default value to -1, and check for -1 instead of 0 in the
src/server.go:startHttpServer() function.

**Comment by junegunn** [comment]
> make it way easier to write scripts that could be run simultaneously

Do you have a strategy for identifying the port assigned to each fzf?  Is there an easy, portable way for accomplishing that?

Also, I'm curious about what you are trying to implement with `--listen`. Can you tell me more about it?

**Comment by czaplicki** [comment]
No clue when it comes for a portable way of getting the port. I really only know about ways in linux (in which there are multiple), since that's the target platform for all my projects. If I can recall correctly, there are ways in macOS and have a hard time  believing there isn't one for BSD. And one would think that at least some of the linux ways would work through WSL, covering windows as well.

I'm using the --listen for somewhat of an hack, as I wanted/needed a way to rebind (as in override) some bindings in runtime, depending on some state.
I'll imagine that the proper way would be to just restart fzf, but wanted to retain all the runtime state. As I really only wanted to update the selected entries.

It's still work in progress, but usable. Just missing a bunch of options/flags and down the line more sources the nerd font, but I'd upload it as a gist in the mean time, if you want to have a look at it^^

https://gist.github.com/czaplicki/684601af9ef02d0491f3bc2aed932ae8

**Comment by junegunn** [comment]
> ways in linux (in which there are multiple)

The overall process feels quite clunky, you first have to know the process id of the target fzf (there can be multiple fzfs running) and use another command to figure out which port is being used by the process. Is it really better than the way you handled it in the linked script? (i.e. `get_avalible_port`) I'm not sure.
